src/github/mod.rs

```diff
@@ -209,34 +209,57 @@ impl<'a> IntoIterator for &'a CachedReleases {
 
 pub struct GitHubApi {
     base_url: String,
+    token: Option<String>,
 }
 
-impl Default for GitHubApi {
+pub struct GitHubApiBuilder {
+    url: String,
+    token: Option<String>,
+}
+
+impl Default for GitHubApiBuilder {
     fn default() -> Self {
         Self::new()
     }
 }
 
-impl GitHubApi {
-    pub fn for_crate<S: Into<String>>(base_url: S) -> Self {
+impl GitHubApiBuilder {
+    pub fn new() -> Self {
         Self {
-            base_url: base_url.into(),
+            url: "https://api.github.com/repos/input-output-hk/jormungandr".into(),
```

Comment:
This could be a `Cow` the way you always initialize it to some string.
This is test code, so it does not matter much.